So just to confirm, we should still expect to see this warning with cms-run3?

`GeoMaterialParams.cc:157: warning: Some geometry volumes do not have known material IDs: world_volume@0x7f7110475c60, cms_OCMS@0x7f7110475bc0, cms_CMSE@0x7f7110475b20, cmsextent_CMStoZDC@0x7f7110474c20, cmsextent_ZDCtoFP4201@0x7f7110474cc0, cmsextent_ZDCtoFP4202@0x7f7110474d60, tracker_Tracker@0x7f7110474e00, trackerpatchpannel_TrackerPatchPannel@0x7f7110474400, trackerpatchpannel_PIXSandwichLayer1outer@0x7f711044a060, ...`

--- a/app/demo-geant-integration/DetectorConstruction.cc
+++ b/app/demo-geant-integration/DetectorConstruction.cc
@@ -57,7 +57,12 @@ G4VPhysicalVolume* DetectorConstruction::Construct()
     CELER_LOG_LOCAL(status) << "Loading detector geometry";
 
     G4GDMLParser gdml_parser;
-    gdml_parser.SetStripFlag(GlobalSetup::Instance()->StripGDMLPointers());
+    gdml_parser.SetStripFlag(true);
+    if (!GlobalSetup::Instance()->StripGDMLPointers())
+    {
+        // DEPRECATED: remove in 1.0?
+        CELER_LOG(warning) << "Ignoring deprecated 'stripGDMLPointers false'";
+    }
 
     std::string const& filename = GlobalSetup::Instance()->GetGeometryFile();
     if (filename.empty())

--- a/app/demo-geant-integration/GlobalSetup.cc
+++ b/app/demo-geant-integration/GlobalSetup.cc
@@ -63,7 +63,7 @@ GlobalSetup::GlobalSetup()
                                                 strip_gdml_pointers_);
         cmd.SetGuidance(
             "Remove pointer suffix from input logical volume names");
-        cmd.SetDefaultValue("false");
+        cmd.SetDefaultValue("true");
     }
     {
         auto& cmd = messenger_->DeclareProperty("outputFile",

--- a/app/demo-geant-integration/GlobalSetup.hh
+++ b/app/demo-geant-integration/GlobalSetup.hh
@@ -73,7 +73,7 @@ class GlobalSetup
     std::string event_file_;
     int root_buffer_size_{128000};
     bool write_sd_hits_{false};
-    bool strip_gdml_pointers_{false};
+    bool strip_gdml_pointers_{true};
     G4ThreeVector field_;
 
     std::unique_ptr<G4GenericMessenger> messenger_;

--- a/src/celeritas/ext/GeantVolumeMapper.cc
+++ b/src/celeritas/ext/GeantVolumeMapper.cc
@@ -55,7 +55,8 @@ VolumeId GeantVolumeMapper::operator()(G4LogicalVolume const& lv) const
     {
         CELER_LOG(warning) << "Failed to exactly match " << celeritas_core_geo
                            << " volume from Geant4 volume '" << lv.GetName()
-                           << "'; found '" << geo.id_to_label(all_ids.front())
+                           << "'@" << static_cast<void const*>(&lv)
+                           << "; found '" << geo.id_to_label(all_ids.front())
                            << "' by omitting the extension";
         return all_ids.front();
     }

--- a/src/celeritas/ext/detail/GeantGeoConverter.cc
+++ b/src/celeritas/ext/detail/GeantGeoConverter.cc
@@ -39,6 +39,7 @@
 #include <G4Polyhedra.hh>
 #include <G4PropagatorInField.hh>
 #include <G4ReflectedSolid.hh>
+#include <G4ReflectionFactory.hh>
 #include <G4RotationMatrix.hh>
 #include <G4Sphere.hh>
 #include <G4SubtractionSolid.hh>
@@ -295,14 +296,14 @@ LogicalVolume* GeantGeoConverter::convert(G4LogicalVolume const* g4_logvol)
 
     LogicalVolume* const vg_logvol
         = new LogicalVolume(clean_name.c_str(), unplaced);
-    logical_volume_map_[g4_logvol] = vg_logvol;
-
-    // Fill helper maps
     VolumeId volid{vg_logvol->id()};
+
+    // Save to helper maps
+    logical_volume_map_[g4_logvol] = vg_logvol;
     g4logvol_id_map_[g4_logvol] = volid;
 
-    // can be used to make a cross check for dimensions and other properties
-    // make a cross check using cubic volume property
+    // Cross check geometry using cubic volume property: very slow for union
+    // solids and a few others
     if (!dynamic_cast<GenericSolidBase const*>(vg_logvol->GetUnplacedVolume())
         && !dynamic_cast<UnplacedScaledShape const*>(
             vg_logvol->GetUnplacedVolume())
@@ -314,13 +315,23 @@ LogicalVolume* GeantGeoConverter::convert(G4LogicalVolume const* g4_logvol)
 
         if (CELER_UNLIKELY(!SoftEqual{0.01}(vg_cap, g4_cap)))
         {
-            CELER_LOG(error)
-                << "Volume " << g4_logvol->GetName() << " (VecGeom volume ID "
-                << volid.get()
+            CELER_LOG(warning)
+                << "Solid type '" << g4_logvol->GetSolid()->GetEntityType()
+                << "' in logical volume '" << g4_logvol->GetName()
+                << "' (VecGeom volume ID " << volid.get()
                 << ") conversion may have failed: VecGeom/G4 volume ratio is "
                 << vg_cap / g4_cap;
         }
     }
+
+    if (auto const* lv = G4ReflectionFactory::Instance()->GetConstituentLV(
+            const_cast<G4LogicalVolume*>(g4_logvol)))
+    {
+        // The *constituent* (unreflected) logical volume is actually tied to
+        // the sensitive detectors: save this as well
+        g4logvol_id_map_[lv] = volid;
+    }
+
     return vg_logvol;
 }
 

--- a/src/celeritas/ext/detail/GeantVolumeVisitor.cc
+++ b/src/celeritas/ext/detail/GeantVolumeVisitor.cc
@@ -74,13 +74,11 @@ void GeantVolumeVisitor::visit(G4LogicalVolume const& logical_volume)
 std::string
 GeantVolumeVisitor::generate_name(G4LogicalVolume const& logical_volume)
 {
-    std::string name = logical_volume.GetName();
-
     // Run the LV through the GDML export name generator so that the volume is
     // uniquely identifiable in VecGeom. Reuse the same instance to reduce
     // overhead: note that the method isn't const correct.
     static G4GDMLWriteStructure temp_writer;
-    return temp_writer.GenerateName(name, &logical_volume);
+    return temp_writer.GenerateName(logical_volume.GetName(), &logical_volume);
 }
 
 //---------------------------------------------------------------------------//

--- a/test/corecel/cont/LabelIdMultiMap.test.cc
+++ b/test/corecel/cont/LabelIdMultiMap.test.cc
@@ -56,7 +56,7 @@ TEST(LabelIdMultiMapTest, empty)
 
 TEST(LabelIdMultiMapTest, no_ext_with_duplicates)
 {
-    CatMultiMap cats{VecLabel{{"dexter", "andy", "loki", "", "", ""}}};
+    CatMultiMap cats{VecLabel{{"dexter", "andy", "loki", "bob", "bob", "bob"}}};
     EXPECT_EQ(6, cats.size());
     EXPECT_EQ(CatId{}, cats.find("nyoka"));
     EXPECT_EQ(CatId{0}, cats.find("dexter"));
@@ -66,6 +66,13 @@ TEST(LabelIdMultiMapTest, no_ext_with_duplicates)
 
     static const CatId expected_duplicates[] = {CatId{3}, CatId{4}, CatId{5}};
     EXPECT_VEC_EQ(expected_duplicates, cats.duplicates());
+}
+
+TEST(LabelIdMultiMapTest, empty_duplicates)
+{
+    CatMultiMap cats{VecLabel{{"dexter", "andy", "loki", "", ""}}};
+    EXPECT_EQ(5, cats.size());
+    EXPECT_EQ(0, cats.duplicates().size());
 }
 
 TEST(LabelIdMultiMapTest, some_labels)

